From 5c3772eaeaf6ea77c8fba723d250c24b606c58da Mon Sep 17 00:00:00 2001
From: Iceows <mounierr07@gmail.com>
Date: Sat, 7 Oct 2023 22:45:21 +0200
Subject: [PATCH 01/13] treble: add mono sim support for dual sim phone

Allows you to switch a dual sim phone to single sim
For huawei phone emui 8 and 9
---
 phh-prop-handler.sh | 31 +++++++++++++++++++++++++++++++
 vndk.rc             |  3 +++
 2 files changed, 34 insertions(+)

diff --git a/phh-prop-handler.sh b/phh-prop-handler.sh
index 2d30ab7..7637cc9 100644
--- a/phh-prop-handler.sh
+++ b/phh-prop-handler.sh
@@ -257,3 +257,34 @@ if [ "$1" == "persist.bluetooth.system_audio_hal.enabled" ]; then
     restartAudio
     exit
 fi
+
+
+#monosim
+if [ "$1" == "persist.sys.phh.monosim" ]; then
+    if [[ "$prop_value" != "false" && "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [ "$prop_value" == "true" ]; then
+		resetprop_phh persist.radio.multisim.config single
+		resetprop_phh persist.dsds.enabled false
+		resetprop_phh persist.radio.modem.cap 09995
+
+		resetprop_phh ro.config.modem_number 1
+		resetprop_phh ro.config.client_number 1
+
+		resetprop_phh ro.multi.rild false
+		resetprop_phh persist.odm.dsds.enabled false
+		resetprop_phh ro.odm.multi.rild false
+
+	else
+		resetprop_phh persist.radio.multisim.config dsds
+		resetprop_phh persist.dsds.enabled true
+		resetprop_phh persist.radio.modem.cap 09A9D
+
+		resetprop_phh ro.multi.rild true
+		resetprop_phh persist.odm.dsds.enabled true
+		resetprop_phh ro.odm.multi.rild true
+    fi
+    exit
+fi
diff --git a/vndk.rc b/vndk.rc
index 823456a..470b9ac 100644
--- a/vndk.rc
+++ b/vndk.rc
@@ -58,6 +58,9 @@ on property:persist.sys.phh.backlight.scale=*
 on property:persist.sys.phh.disable_soundvolume_effect=*
     exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.disable_soundvolume_effect"
 
+on property:persist.sys.phh.monosim=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.monosim"
+
 on property:ro.sf.lcd_density=*
     restart surfaceflinger
 
-- 
2.33.0.windows.2

